src/welcomePage/WelcomePageOnboarding/WelcomePageOnboarding.test.tsx

```diff
@@ -0,0 +1,231 @@
+import { useUser } from "@dashboard/auth";
+import { ApolloMockedProvider } from "@test/ApolloMockedProvider";
+import { render, waitFor } from "@testing-library/react";
+import React from "react";
+
+import { onboardingCompletedMock, onboardingInitState } from "./mocks";
+import { OnboardingProvider } from "./onboardingContext";
+import { useOnboardingStorage } from "./onboardingContext/useOnboardingStorage";
+import { WelcomePageOnboarding } from "./WelcomePageOnboarding";
+
+jest.mock("@dashboard/auth");
+jest.mock("@dashboard/featureFlags", () => ({
+  useFlag: jest.fn().mockReturnValue({
+    enabled: true,
+  }),
+}));
+jest.mock("react-intl", () => ({
+  useIntl: jest.fn(() => ({
+    formatMessage: jest.fn(x => x.defaultMessage),
+  })),
+  defineMessages: jest.fn(x => x),
+  FormattedMessage: ({ defaultMessage }: { defaultMessage: any }) => <>{defaultMessage}</>,
+}));
+jest.mock("./onboardingContext/useOnboardingStorage");
+jest.useFakeTimers();
+jest.mock("@dashboard/components/DevModePanel/hooks", () => ({
+  useDevModeContext: jest.fn(),
+}));
+jest.mock("react-router-dom", () => ({
+  Link: jest.fn(({ to, ...props }) => <a href={to} {...props} />),
+}));
+
+const Wrapper = ({ children }: { children: React.ReactNode }) => (
+  <ApolloMockedProvider>
+    <OnboardingProvider>{children}</OnboardingProvider>
+  </ApolloMockedProvider>
+);
+
+const OLD_ACCOUNT_DATE = "2022-12-31";
+const NEW_ACCOUNT_DATE = "2024-12-31";
+const onboardingCompleteMessage = "Onboarding completed 🎉";
+const allMarkAsDoneStepsIds = [
+  "create-product",
+  "explore-orders",
+  "graphql-playground",
+  "view-webhooks",
+  "invite-staff",
+];
+
+describe("WelcomePageOnboarding", () => {
+  it("should show 'Onboarding completed' when user's account is older than set date", () => {
+    // Arrange
+    (useUser as jest.Mock).mockReturnValue({ user: { dateJoined: OLD_ACCOUNT_DATE } });
+    (useOnboardingStorage as jest.Mock).mockReturnValue({
+      getOnboardingState: jest.fn(() => onboardingInitState),
+      saveOnboardingState: jest.fn(),
+    });
+
+    // Act
+    const { getByText } = render(
+      <Wrapper>
+        <WelcomePageOnboarding />
+      </Wrapper>,
+    );
+
+    // Assert
+    expect(getByText(onboardingCompleteMessage)).toBeInTheDocument();
```

Comment:
Using screen is recommended way to query elements
```suggestion
    render(
      <Wrapper>
        <WelcomePageOnboarding />
      </Wrapper>,
    );

    // Assert
    expect(screen.getByText(onboardingCompleteMessage)).toBeInTheDocument();
```